### PR TITLE
deprecate mock methods to avoid warnings

### LINKED ISSUE #3453: Deprecate methods in  MockHttpServletResponse
To avoid warnings these methods should be marked deprecated:

```
  @Override @Deprecated
  public String encodeRedirectUrl(String arg0) {
    return null;
  }

  @Override @Deprecated
  public String encodeUrl(String arg0) {
    return null;
  }

  @Override @Deprecated
  public void setStatus(int arg0, String arg1) {
  }
```

### PR DESCRIPTION
**Pull request by elharo**
fix #3453 @briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl [hunk 1]
```diff
@@ -147,31 +147,31 @@ public class MockHttpServletResponse implements HttpServletResponse {
   }
 
   @Override
-  public void sendError(int arg0, String arg1) throws IOException {
+  public void sendError(int stausCode, String message) throws IOException {
   }
 
   @Override
   public void sendRedirect(String arg0) throws IOException {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: `arg0` to (probably) `url`?

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl [hunk 2]
```diff
@@ -147,31 +147,31 @@ public class MockHttpServletResponse implements HttpServletResponse {
   }
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: Above this methods, `sendError(int arg0)` --> `sendError(int statusCode`).

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl [hunk 3]
```diff
@@ -3,8 +3,8 @@
 </#if>import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
-import java.util.Collection;
-import java.util.Locale;
+<#if servletVersion != "2.5"> import java.util.Collection;
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: There's a blank before `import java...`.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: removed

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: done

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl (hunk 2 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: This is still not fixed. There's a method with `arg0` in the file.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.util/templates/appengine/MockHttpServletResponse.java.ftl]: done

**Pull request merged**